docs/streams/stream-basics/producers.md

```diff
@@ -23,4 +23,52 @@ Messages published by producers can be compressed during transportation in order
+
+With message chunking enabled, when the size of a message exceeds the allowed maximum payload size (the `maxMessageSize` parameter of the broker), the workflow of messaging is as follows:
+
+1. The producer splits the original message into chunked messages and publishes them with chunked metadata to the broker separately and in order.
+2. The broker stores the chunked messages in one managed ledger in the same way as that of ordinary messages, and it uses the chunkedMessageRate parameter to record chunked message rate on the topic.
+3. The consumer buffers the chunked messages and aggregates them into the receiver queue when it receives all the chunks of a message.
+4. The client consumes the aggregated message from the receiver queue.
+
+:::note
+
+- Chunking is only available for persistent topics.
+- Chunking cannot be enabled simultaneously with batching. Before enabling chunking, you must turn off batching.
+
+:::
+
+### Handle Consecutive Chunked Messages with One Ordered Consumer
+
+The following figure shows a topic with one producer that publishes a large message payload in chunked messages along with regular non-chunked messages.
+
+The producer publishes message M1 in three chunks labeled M1-C1, M1-C2, and M1-C3. The broker stores all the three chunked messages in the managed ledger and dispatches them to the ordered (exclusive/failover) consumer in the same order.
+
+The consumer buffers all the chunked messages in memory until it receives all the chunked messages, aggregates them into one message and then hands over the original message M1 to the client.
+
+![Consecutive chunked message diagram](/img/streams/chunking1.png)
+
+### Handle Interwoven Chunked Messages with One Ordered Consumer
+
+When multiple producers publish chunked messages into a single topic, the broker stores all the chunked messages coming from different producers in the same managed ledger. The chunked messages in the managed ledger can be interwoven with each other.
+
+As shown below, Producer 1 publishes message M1 in three chunks M1-C1, M1-C2 and M1-C3. Producer 2 publishes message M2 in three chunks M2-C1, M2-C2 and M2-C3. All chunked messages of the specific message are still in order but might not be consecutive in the managed ledger.
+
+![Interwoven chunked message diagram](/img/streams/chunking2.png)
+
+In this case, interwoven chunked messages might bring some memory pressure to the consumer because the consumer keeps a separate buffer for each large message to aggregate all its chunks in one message. You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `maxPendingChunkedMessage` parameter. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later, optimizing memory utilization.
```

Comment:
```suggestion
```

@diana-macrometa - We do not need this information in my opinion. It seems too much.